### DISCUSSION TITLE
[ZK Stack] - Command failed to run: forge script deploy-scripts/DeployL2Contracts.sol when trying to deploy on Holesky testnet

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#685] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [x] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I was following this tutorial https://code.zksync.io/tutorials/custom-zk-chain to deploy zk chain to holesky but Im stuck on this error. Local L1 ETH is working fine and I have no issues with in terms of deploying contracts.
We are using chainstack enterprise edition RPC, so I believe there is no issue with the RPC server.

Error: 
script failed: custom error e450d38c:000000000000000000000000db76e8ab…000000000000000000a864b43118f400 (96 bytes)
│
■  Command failed to run: forge script deploy-scripts/DeployL2Contracts.sol --legacy --ffi --rpc-url=https://ethereum-holesky.core.chainstack.com/xxxxxxxxxxxxxxxxxxxxxx --broadcast --private-key=ced40ebef6bc47dfedc38244cfaa7605e54890dc8dbc85bt9bbaa4e175707822

### Expected Behavior

Contract should get deployed

### Current Behavior

Failed to deploy DeployL2Contracts.sol

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It appears you are encountering a deployment issue on the Holsky testnet using the Forge script. The error message you provided suggests a custom error during the script execution, which might be related to the specific contract or the deployment environment.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Verify Contract Compatibility**: Ensure that the contract `DeployL2Contracts.sol` is compatible with the Holsky testnet specifications and the Solidity version you are using.

2. **Check the RPC Connection**: Although you mentioned using Chainstack Enterprise Edition RPC, it's good to double-check if there are any restrictions or specific configurations required by Chainstack that might affect contract deployment.

3. **Review the Deployment Script**: Look into the `forge script` command options you are using, especially the `--legacy` and `--ffi` flags, to ensure they are appropriate for your deployment scenario.

4. **Consult Documentation**: Since you are following a specific tutorial, review the steps in the tutorial to ensure all prerequisites and configurations are correctly set up. Here is the link to the tutorial for reference: [Custom ZK Chain Tutorial](https://code.zksync.io/tutorials/custom-zk-chain).

5. **Error Logs and Debugging**: If possible, enable more verbose logging for the deployment process to capture more detailed error information. This might provide further insights into what might be causing the deployment failure.

If after these steps the issue persists, consider reaching out for more specific support or consulting the community or forums related to the tools and platforms you are using.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by sarahschwartz** [comment]
Hi @sachanayush47, were you able to go through the tutorial as-is without any issues?

**Reply by bxpana** [reply]
Also noticed that your `--private-key` contains a `t` which might be causing the issue with deploying
`ced40ebef6bc47dfedc38244cfaa7605e54890dc8dbc85b t 9bbaa4e175707822`

**Reply by sachanayush47** [reply]
> Also noticed that your `--private-key` contains a `t` which might be causing the issue with deploying
> `ced40ebef6bc47dfedc38244cfaa7605e54890dc8dbc85b t 9bbaa4e175707822`

I did that intentionally to keep the private key private before posting on GitHub discussion.

**Reply by bxpana** [reply]
Nice, ok. Will see if there would be any reason why it wouldn't work.

**Comment by sachanayush47** [comment]
> Hi @sachanayush47, were you able to go through the tutorial as-is without any issues?

Hi Sarah,
Yeah Im able to go through the tutorial without any issues. Issues are only there when trying to deploy holesky.

**Comment by sachanayush47** [comment, marked as answer]
Hi, The issue has been fixed. It was my mistake. I was using custom token and forgot to fund the governor's address with custom token.

**Reply by bxpana** [reply]
Appreciate the update and glad you were able to get it working.
